This patch allows multi-line SQL scripts, executed with tg-admin sql
upgrade, to complete if your backend database is sqlite.

filed upstream as
http://sourceforge.net/tracker/?func=detail&aid=2883211&group_id=74338&atid=540674


--- sqlobject/sqlite/sqliteconnection.py.~1~	2008-03-07 09:54:30.000000000 -0600
+++ sqlobject/sqlite/sqliteconnection.py	2009-10-21 08:03:40.000000000 -0500
@@ -172,7 +172,10 @@
         if self.debug:
             self.printDebug(conn, query, 'QueryR')
         try:
-            return cursor.execute(query)
+            if query.count('\n') <= 1:
+                return cursor.execute(query)
+            else:
+                return cursor.executescript(query)
         except self.module.OperationalError, e:
             raise OperationalError(ErrorMessage(e))
         except self.module.IntegrityError, e:
